pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1100,6 +1105,94 @@ func (r ReconcileHyperConverged) applyUpgradePatch(req *common.HcoRequest, hcoJs
 	return hcoJson, nil
 }
 
+func (r ReconcileHyperConverged) removeOldMetricsObjs(req *common.HcoRequest) (bool, error) {
+	anyServiceRemoved, err := r.removeObjs(req, corev1.Service{}, []string{
+		"hyperconverged-cluster-operator-metrics",
+		"hyperconverged-cluster-webhook-metrics",
+	})
+	if err != nil {
+		return false, err
+	}
+
+	anyEndpointRemoved, err := r.removeObjs(req, corev1.Endpoints{}, []string{
+		"hyperconverged-cluster-operator-metrics",
+	})
+	if err != nil {
+		return false, err
+	}
+
+	return anyServiceRemoved || anyEndpointRemoved, nil
+}
+
+func (r ReconcileHyperConverged) removeObjs(req *common.HcoRequest, objType interface{}, objNames []string) (bool, error) {
+	anyRemoved := false
+
+	for _, objName := range objNames {
+		removed, err := r.removeObj(req, objType, objName)
+		if err != nil {
+			return false, err
+		}
+		anyRemoved = anyRemoved || removed
+	}
+
+	return anyRemoved, nil
+}
+
+func (r ReconcileHyperConverged) removeObj(req *common.HcoRequest, objType interface{}, objName string) (bool, error) {
+	obj, err := r.getObj(req, objType, objName)
+	if err != nil {
+		return false, err
+	} else if obj == nil {
+		return false, nil
+	}
+
```

Comment:
I think `hcoutil.EnsureDeleted` already does this logic. Please check if we can remove these lines and the `getObj` function.